cds_ils/migrator/eitems/api.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2019-2020 CERN.
+#
+# CDS-ILS is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-ILS migrator API."""
+
+import io
+import uuid
+from io import BytesIO
+
+import click
+import requests
+from invenio_app_ils.documents.api import Document
+from invenio_app_ils.documents.indexer import DocumentIndexer
+from invenio_app_ils.eitems.api import EItem, EItemIdProvider
+from invenio_app_ils.eitems.indexer import EItemIndexer
+from invenio_db import db
+from invenio_files_rest.models import Bucket, ObjectVersion
+
+from cds_ils.migrator.documents.api import get_all_documents_with_files
+
+
+def import_legacy_files(file_link):
+    """Download file from legacy."""
+    return io.BytesIO(requests.get(file_link, stream=True).content)
+
+
+def create_file(bucket, file_stream, filename):
+    """Create file for given bucket."""
+    file_record = ObjectVersion.create(bucket, filename, stream=file_stream)
+    db.session.add(file_record)
+    db.session.commit()
+    return file_record
+
+
+def create_eitem_with_bucket_for_document(document_pid):
+    """Create EItem record."""
+    obj = {"document_pid": document_pid, "open_access": True}
+    record_uuid = uuid.uuid4()
+    provider = EItemIdProvider.create(
+        object_type="rec",
+        object_uuid=record_uuid,
+    )
+
+    obj["pid"] = provider.pid.pid_value
+    with db.session.begin_nested():
+        eitem = EItem.create(obj, record_uuid)
+        eitem.commit()
+        bucket = Bucket.create()
+        eitem["bucket_id"] = str(bucket.id)
+        eitem.commit()
+    db.session.commit()
+    return eitem, bucket
+
+
+def process_files_from_legacy():
+    """Process legacy file."""
+    results = get_all_documents_with_files()
+    click.echo(
+        "Found {} documents with files.".format(results.hits.total.value)
+    )
+    for hit in results:
+
+        # make sure the document is in DB not only ES
+        document = Document.get_record_by_pid(hit.pid)
+        click.echo("Processing document {}...".format(document["pid"]))
+        for file_link in document["_migration"]["eitems_file_links"]:
+
+            click.echo("File: {}".format(file_link))
+            eitem, bucket = create_eitem_with_bucket_for_document(
```

Comment:
what happens if the download fails? I would maybe first download the file, then add it there?
can we get the checksum somehow to be sure that we have a file that is intact?

Reply:
I was thinking about the checksum, I can add it